### PR TITLE
VideoPlayer: ffmpeg - drop deprecated merging of side data

### PR DESCRIPTION
**Pull request by fritsch**
Fixes playback of certain TrueHD files.

### CODE REVIEW

**Comment by fritsch** [comment]
Thanks @FernetMenta 

@MartijnKaijser for you to test. Works fine for me with the samples we checked yesterday.

**Comment by fritsch** [comment]
Wait for merging, there is a "minor" segfault involved currently: http://paste.ubuntu.com/25896241/

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxUtils.cpp [hunk 1]
```diff
@@ -36,13 +36,17 @@ void CDVDDemuxUtils::FreeDemuxPacket(DemuxPacket* pPacket)
...
-      CLog::Log(LOGERROR, "%s - Exception thrown while freeing packet", __FUNCTION__);
+    if (pPacket->pData)
+      _aligned_free(pPacket->pData);
+    if (pPacket->iSideDataElems)
+    {
+      AVPacket avPkt;
+      av_init_packet(&avPkt);
+      avPkt.side_data = reinterpret_cast<AVPacketSideData*>(pPacket->pSideData);
+      avPkt.side_data_elems = pPacket->iSideDataElems;
+      av_packet_free_side_data(&avPkt);
```
Comment (fritsch) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxUtils.cpp]: Don't we have to set pPacket->iSideDatalElems to 0 too?

Comment (fritsch) [reply xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxUtils.cpp]: and src ptr might be null - which directly segfaults: https://ffmpeg.org/doxygen/3.1/avpacket_8c_source.html#l00531

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxUtils.cpp [hunk 2]
```diff
@@ -82,3 +86,12 @@ DemuxPacket* CDVDDemuxUtils::AllocateDemuxPacket(unsigned int iDataSize, unsigne
     ret->cryptoInfo = std::shared_ptr<DemuxCryptoInfo>(new DemuxCryptoInfo(encryptedSubsampleCount));
   return ret;
 }
+
+void CDVDDemuxUtils::StoreSideData(DemuxPacket *pkt, AVPacket *src)
+{
+  AVPacket avPkt;
+  av_init_packet(&avPkt);
+  av_packet_copy_props(&avPkt, src);
```
Comment (fritsch) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxUtils.cpp]: src might be null, which directly crashes: https://ffmpeg.org/doxygen/3.1/avpacket_8c_source.html#l00531

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxUtils.cpp (hunk 1 above)
Comment (fritsch) [reply xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxUtils.cpp]: I think the reinterpret_cast is wrong here and we end up with a dangeling ptr.

**Comment by fritsch** [comment]
Ha :-) I missed to rebuild vnsi ... all good.

**Comment by fritsch** [comment]
jenkins build this please

**Comment by fritsch** [comment]
I changed the reinterpret_cast to static_cast

**Review by FernetMenta — APPROVED** [review]

**Comment by FernetMenta** [comment]
The ugly thing is that pvr and inputstream addons expose DemuxPacket via their own APIs. Both versions should be bumped.

**Comment by MartijnKaijser** [comment]
@fritsch ping

**Comment by fritsch** [comment]
As a simple rebuild is enough - I hessitate a bit to bump version as we are mid in development.

@FernetMenta what do you think is the right way forward?

**Comment by FernetMenta** [comment]
I don't understand your concerns. We have bumped versions always for things like that.

**Comment by fritsch** [comment]
My concerns are: API is not changed, but ABI - we want to trigger a rebuild, nothing else.

**Comment by FernetMenta** [comment]
API is changed because the header included by an addon is changed. This is per definition an API change that should be reflected by a version bump.

**Comment by fritsch** [comment]
I think what I realy don't like is that we "govern" a hidden API here :-) - but never the less, bumped min version and current version of inputstream and pvr.

**Comment by fritsch** [comment]
@FernetMenta good to go - anyone else I need to inform?

**Comment by FernetMenta** [comment]
good to go

**Comment by ksooo** [comment]
@MilhouseVH please be aware that this PR requires a recompile of all PVR and inputstream addons.

**Comment by MilhouseVH** [comment]
@ksooo thanks - I rebuild all the addons I include in the release every night. Those I don't include...oh well! :)

**Comment by wsnipex** [comment]
just FYI. ALL addons that need to be rebuilt, should have their version bumped, otherwise no PPA updates..

**Comment by fritsch** [comment]
That's the reason this PR stood here for more than 4 weeks - seems it did not work as intended. Do we have a list? Most likely everything inputstream and pvr related that use the packet.

**Pull request merged**